### PR TITLE
Add sensitivity toggles in filter sidebar and modal

### LINKED ISSUE #2368: Component: Sensitive content fetch/blur sidebar toggles
## Description

Define the component that contains two toggles, one to fetch the sensitive content from the API and the other that controls whether the sensitive content fetched should be blurred in the UI.

## API

This component need not accept props or emit events, it can be directly synced with the state held in the Pinia store.

### State

| Name | Type | Description |
|-|-|-|
| shouldFetchSensitive | boolean | whether sensitive content should be fetched from the API |
| shouldBlurSensitive | boolean | whether sensitive content should be blurred in the search results and single result page |

`shouldFetchSensitive` is synced with the feature flag store, specifically the [`fetch_sensitive` feature flag](https://github.com/WordPress/openverse/blob/de7d99e70bd56df8552608a9edc0213821068089/frontend/feat/feature-flags.json#LL36C6-L36C21).

`shouldBlurSensitive` can be stored in the `ui` store but it should not be persisted across visits, so it cannot be committed to a cookie or put into `localStorage`.

> **Note**
> The second toggle should be turned off **and** disabled if the former is off.

## Screenshot

Screenshot 2023-06-09 at 9 13 42 AM

## Figma 

- **Figma:** https://www.figma.com/file/bDTfGzBit8rGRPktOR1cu4/Safe-content-browsing-flow?type=design&node-id=1140-23930&t=t4b5KNE2t2viAres-4

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #2368 by @dhruvkb

## Description

This PR adds toggles in the sidebar to control the fetching and blurring of sensitive content.

The next step for the project would be to link the first toggle with the search store's filter mechanism so that changing it applies the right query params in the search request, i.e. #2548

## Testing Instructions

0. Check out the PR and run the dev server.
1. Enable `fake_sensitive` and `sensitive_content` flags. The former ensures there is some content to test blurring with and the latter controls the visibility of the toggles.
2. Search for something.
3. Open the filter sidebar, you should see the toggles.
4. The first toggle controls the `fetch_sensitive` feature flag. Toggle it to see the session storage value change.
5. The second toggle controls whether sensitive content appears blurred. Toggle it to blur/unblur the search results.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/components/VAudioThumbnail/VAudioThumbnail.vue [hunk 1]
```diff
@@ -25,7 +25,7 @@
     <div v-show="audio.thumbnail && isOk" class="thumbnail absolute inset-0">
       <img
         ref="imgEl"
-        class="h-full w-full overflow-clip object-cover object-center"
+        class="h-full w-full overflow-clip object-cover object-center duration-200 motion-safe:transition-[filter,transform]"
```
Comment (zackkrida) [inline frontend/src/components/VAudioThumbnail/VAudioThumbnail.vue:28]: `motion-safe:` is _awesome_, TIL.

File: frontend/src/components/VHeader/VHeaderMobile/VContentSettingsModalContent.vue [hunk 2]
```diff
@@ -49,11 +49,17 @@
...
+      <!-- Horizontal padding removed to display divider. -->
+      <VTabPanel v-if="showFilters" id="filters" class="px-0">
         <VSearchGridFilter
+          class="px-6"
           :show-filter-header="false"
           :change-tab-order="false"
         />
+        <VSafeBrowsing
+          v-if="isSensitiveContentEnabled"
+          class="border-t border-dark-charcoal-20 px-6 pt-6"
```
Comment (zackkrida) [inline frontend/src/components/VHeader/VHeaderMobile/VContentSettingsModalContent.vue:61]: Nice to add these styles in their sidebar, instead of hardcoding them into the component 👍

File: frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue [hunk 3]
```diff
@@ -0,0 +1,110 @@
+<template>
+  <section id="safe-browsing" aria-labelledby="safe-browsing-heading">
+    <header class="relative mb-6 flex items-center justify-between">
+      <h4 id="safe-browsing-headers" class="caption-bold uppercase">
```
Comment (zackkrida) [inline frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue:4]:
```suggestion
      <h4 id="safe-browsing-heading" class="caption-bold uppercase">
```

Typo, I believe

File: frontend/src/components/VFilters/VSearchGridFilter.vue [hunk 4]
```diff
@@ -74,9 +73,6 @@ export default defineComponent({
       default: true,
     },
   },
-  emits: {
-    close: defineEvent(),
-  },
```
Comment (AetherUnbound) [inline frontend/src/components/VFilters/VSearchGridFilter.vue:79]: Why was this removed?

**Review by zackkrida — APPROVED** [review]
I found one typo which concerns accessibility but otherwise LGTM.

File: frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      </h4>
+    </header>
+    <i18n tag="p" path="filters.safeBrowsing.desc" class="label-regular">
+      <template #sensitive>
+        <VLink :href="sensitivityPath">{{
+          $t("filters.safeBrowsing.sensitive")
+        }}</VLink>
+      </template>
+    </i18n>
+    <form class="safe-browsing-form">
```
Comment (AetherUnbound) [inline frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue:15]: This is more of a pedagogical question, but why do we have something like `VLink` for links, presumably, but not something like `VForm` for more generic forms? Is it just because we have common functionality we want for links, but not for forms?

File: frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+ * that may contain sensitive content and subsequently blurring them to prevent
+ * accidental exposure or showing them directly.
+ */
+export default defineComponent({
+  name: "VSafeBrowsing",
+  components: { VCheckbox, VLink },
+  setup() {
+    const { app } = useContext()
+
+    const sensitivityPath = computed(() => app.localePath("/about")) // TODO
```
Comment (AetherUnbound) [inline frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue:65]: Do we have an issue associated with this? Should it be linked?

File: frontend/src/locales/scripts/en.json5 [hunk 7]
```diff
@@ -337,6 +337,21 @@
...
+      desc: "Content marked as {sensitive} is not shown by default.",
+      sensitive: "sensitive",
+      toggles: {
+        fetchSensitive: {
+          title: "Sensitive results",
+          desc: "Show results marked as sensitive in the results area.",
+        },
+        blurSensitive: {
+          title: "Blurred content",
+          desc: "Images and texts are blurred to prevent seeing them without consent.",
```
Comment (AetherUnbound) [inline frontend/src/locales/scripts/en.json5:351]: Whereas the previous toggle describes an action, this one just makes a statement. Should it be something like `Blur content` rather than `Blurred content`, and say something like "Blur images and text to prevent seeing sensitive material."?

**Review by AetherUnbound — APPROVED** [review]
This looks great and works as-expected locally! I have a few mostly questions, one comment on phrasing but nothing to block a merge 🚀

File: frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue (hunk 5 above)
Comment (zackkrida) [reply frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue:15]: > Is it just because we have common functionality we want for links, but not for forms?

Basically, yes. We have very few forms on the site so haven't needed a reusable abstraction. Also `form` elements are a non-visual wrapper for user inputs and don't really have any reusable functionality we need to encapsulate in our use case. This form, for example, only has a css class applied, so switching to `<VForm class="safe-browsing-form">` if it existed wouldn't offer any benefit.

File: frontend/src/locales/scripts/en.json5 (hunk 7 above)
Comment (zackkrida) [reply frontend/src/locales/scripts/en.json5:351]: Agreed. I also don't think "Blurred" will necessarily translate well.

File: frontend/src/components/VFilters/VSearchGridFilter.vue (hunk 4 above)
Comment (dhruvkb) [reply frontend/src/components/VFilters/VSearchGridFilter.vue:79]: `emits` is supposed to document and set types for events emitted by a component but this event wasn't actually emitted anywhere.

File: frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue (hunk 6 above)
Comment (dhruvkb) [reply frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue:65]: Yes, there is an issue, #2550. I'll update the `// TODO` to link to that issue.

File: frontend/src/locales/scripts/en.json5 (hunk 7 above)
Comment (dhruvkb) [reply frontend/src/locales/scripts/en.json5]: Fixed in 543d2b9.

File: frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue (hunk 6 above)
Comment (dhruvkb) [reply frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue]: Fixed in 63c2167.

File: frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue (hunk 5 above)
Comment (AetherUnbound) [reply frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue:15]: Ahh, I see! Thank you!
